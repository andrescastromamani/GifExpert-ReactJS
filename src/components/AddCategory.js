import React, { useState } from 'react'
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategories }) => {
    const [inputValue, setInputValue] = useState('');
    const handleAddCategory = (e) => {
        setInputValue(e.target.value);
    }
    const handleSubmit = (e) => {
        e.preventDefault();
        if (inputValue.trim().length > 0) {
            setCategories(cats => [inputValue, ...cats])
            setInputValue('');
        }
    }
    return (
        <form onSubmit={handleSubmit}>
            <p>{inputValue}</p>
            <input
                type="text"
                placeholder={inputValue}
                value={inputValue}
                onChange={handleAddCategory} />
        </form>
    )
}
AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}