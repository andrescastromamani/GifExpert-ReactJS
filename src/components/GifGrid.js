import React, { useEffect, useState } from "react";
import { useFetchGifs } from "../hooks/useFetchGifs";
//import { getGifs } from '../helpers/getGifs'
import { GifGridItem } from "./GifGridItem";

export const GifGrid = ({ category }) => {
    /*const [images, setImages] = useState([])
    useEffect(() => {
        getGifs(category).then(setImages)
    }, [category])*/
    const { loading, data } = useFetchGifs();
    console.log(data);
    return (
        <>
            <h3>{category}</h3>
            <div className="card-grid">
                {
                    loading ? 'Loading....' : 'Listo'
                }
                {/*
                    images.map(img => (
                        <GifGridItem
                            key={img.id}
                            {...img}
                        />
                    ))*/
                }
            </div>
        </>
    )
}