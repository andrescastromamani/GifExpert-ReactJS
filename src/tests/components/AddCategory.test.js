import React from "react";
import { shallow } from "enzyme";
import { AddCategory } from "../../components/AddCategory";

describe('AddCategory', () => {
    const setCategories = jest.fn();
    let wrapper = shallow(<AddCategory setCategories={setCategories} />);

    beforeEach(() => {
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories={setCategories} />);
    })

    test('should render correctly', () => {
        expect(wrapper).toMatchSnapshot();
    })
    test('should call setCategories and add the new category', () => {
        const input = wrapper.find('input');
        const value = 'Hola';
        input.simulate('change', { target: { value } });
        expect(wrapper.find('p').text()).toBe(value);
    })
    test('Submit form', () => {
        wrapper.find('form').simulate('submit', { preventDefault() { } });
        expect(setCategories).not.toHaveBeenCalled();
    })
    test('should executed setCategories and  clear the input', () => {
        const value = 'Hola';
        wrapper.find('input').simulate('change', { target: { value } });
        wrapper.find('form').simulate('submit', { preventDefault() { } });
        expect(setCategories).toHaveBeenCalled();
        expect(setCategories).toHaveBeenCalledWith(expect.any(Function));

        //Verify clear input
        expect(wrapper.find('input').prop('value')).toBe('');
    })
})