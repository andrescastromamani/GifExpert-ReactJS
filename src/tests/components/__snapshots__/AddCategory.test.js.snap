// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCategory should render correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    placeholder=""
    type="text"
    value=""
  />
</form>
`;
